src/site/content/en/blog/fast-playback-with-preload/index.md

```diff
@@ -0,0 +1,604 @@
+      // network response in parallel as well.
+    }
+
+    // Browser sends a HTTP Range request. Let's provide one reconstructed
+    // manually from the cache.
+    if (request.headers.has('range')) {
+      return response.blob()
+      .then(data => {
+
+        // Get start position from Range request header.
+        const pos = Number(/^bytes\=(\d+)\-/g.exec(request.headers.get('range'))[1]);
+        const options = {
+          status: 206,
+          statusText: 'Partial Content',
+          headers: response.headers
+        }
+        const slicedResponse = new Response(data.slice(pos), options);
+        slicedResponse.setHeaders('Content-Range': 'bytes ' + pos + '-' +
+            (data.size - 1) + '/' + data.size);
+        slicedResponse.setHeaders('X-From-Cache': 'true');
+
+        return slicedResponse;
+      });
+    }
+
+    return response;
+  }
+}
+```
+
+It is important to note that I used `response.blob()` to recreate this sliced
+response as this simply gives me a handle to the file while
+`response.arrayBuffer()` brings the entire file into renderer memory.
+
+My custom `X-From-Cache` HTTP header can be used to know whether this request
+came from the cache or from the network. It can be used by a player such as
+[ShakaPlayer][shakaplayer] to ignore the response time as an indicator of
+network speed.
+
+{% YouTube 'f8EGZa32Mts' %}
+
+Have a look at the official [Sample Media App][sample-media-app] and in particular its
+[ranged-response.js] file for a complete solution for how to handle `Range`
+requests.
+
+
+[how much information or content to preload]: https://web.dev/video-and-source-tags/#preload
+[video preload attribute]: https://web.dev/video-and-source-tags/#preload
+[chrome64]: /web/updates/2017/12/chrome-63-64-media-updates#media-preload-defaults-metadata
+[datasaver]: https://support.google.com/chrome/answer/2392284
+[covered]: https://developers.google.com/web/updates/2016/03/link-rel-preload
+[articles]: https://www.smashingmagazine.com/2016/02/preload-what-is-it-good-for/
+[link preload]: https://w3c.github.io/preload/
+[cache]: https://web.dev/storage-for-the-web/
+[shakaplayer]: https://github.com/google/shaka-player/blob/master/docs/tutorials/service-worker.md
+[sample-media-app]: https://github.com/GoogleChrome/sample-media-pwa
+[ranged-response.js]: https://github.com/GoogleChrome/sample-media-pwa/blob/master/src/client/scripts/ranged-response.js
+[cache-api]: https://developer.mozilla.org/en-US/docs/Web/API/Cache
+[yet]: https://github.com/whatwg/fetch/issues/144
+[shaka-player]: https://github.com/google/shaka-player
+[jw-player]: https://developer.jwplayer.com/
+[video-js]: http://videojs.com/
+[network-info-sample]: https://googlechrome.github.io/samples/network-information/
+[fast-light]: https://developers.google.com/web/updates/2016/02/save-data
+[mse-basics]: https://developers.google.com/web/fundamentals/media/mse/basics
```

Comment:
Link is 404 because of https://github.com/google/WebFundamentals/pull/9053/

Reply:
I'll work on these next. In the meantime, I've hidden or replace the links.